### PR TITLE
[fix] piratebay engine

### LINKED ISSUE #1935: thepiratebay engine is broken
When trying "piratebay" engine on an old version of thepiratebay (because the new version returns no results) I get this parse error:
```
ERROR:searx.search:engine piratebay : exception : list index out of range
Traceback (most recent call last):
  File "/tmp/searx/searx/search.py", line 156, in search_one_http_request_safe
    search_results = search_one_http_request(engine, query, request_params)
  File "/tmp/searx/searx/search.py", line 97, in search_one_http_request
    return engine.response(response)
  File "/tmp/searx/searx/engines/piratebay.py", line 60, in response
    link = result.xpath('.//div[@class="detName"]//a')[0]
IndexError: list index out of range
```

The pirate bay proxy used: https:// pirateproxy dot live (the URL is intentionally broken because I don't want to directly share website with potential "pirated content" on the repository).

### PR DESCRIPTION
**Pull request by unixfox**
## What does this PR do?

This PR fixes the piratebay engine.

## Why is this change important?

The piratebay engine was broken.

## How to test this PR locally?

1. Run Searx locally
2. Type `!tpb test` in the search bar.

## Related issues

Closes #1935

### CODE REVIEW

**Review by asciimoo — APPROVED** [review]
Great, thanks!

File: searx/engines/piratebay.py [hunk 1]
```diff
@@ -1,44 +1,50 @@
 #  Piratebay (Videos, Music, Files)
 #
-# @website     https://thepiratebay.se
-# @provide-api no (nothing found)
+# @website     https://thepiratebay.org
+# @provide-api yes (https://apibay.org/)
```
Comment (SuperSandro2000) [inline searx/engines/piratebay.py:4]: returns 403 for me

Comment (unixfox) [reply searx/engines/piratebay.py:4]: But not for the other endpoints that the engine use.
Go to thepiratebay.org and look at the network activity you will understand how the API is used.

File: searx/engines/piratebay.py [hunk 2]
```diff
@@ -47,50 +53,43 @@ def request(query, params):
...
-
-        magnetlink = result.xpath(magnet_xpath)[0]
-        torrentfile_links = result.xpath(torrent_xpath)
-        if torrentfile_links:
-            torrentfile_link = torrentfile_links[0].attrib.get('href')
-        else:
-            torrentfile_link = None
+    for result in search_res:
+        link = url + "description.php?id=" + result["id"]
+        magnetlink = "magnet:?xt=urn:btih:" + result["info_hash"] + "&dn=" + result["name"]
```
Comment (dalf) [inline searx/engines/piratebay.py:65]: Would it better with a ```\``` at the end of the line ?

Comment (unixfox) [reply searx/engines/piratebay.py:65]: @dalf That's why I opened a PR in order to fix that: https://github.com/asciimoo/searx/pull/2134 and I'm waiting for someone to merge it :)

**Pull request merged**